src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1635,6 +1639,40 @@ else if (null == currentPrimaryPlaceHolder) {
                     currentConnectPlaceHolder = currentPrimaryPlaceHolder;
                 }
 
+                try {
+                    InetAddress[] inetAddrs = InetAddress.getAllByName(currentConnectPlaceHolder.getServerName());
+
+                    // if only one IP is resolved and user didn't set TNIR explicitly then set TNIR to false
+                    if ((1 == inetAddrs.length) && !userSetTNIR) {
```

Comment:
Oh, sorry, I missed this comment. If that's the case, I guess I need to roll back a lot of changes I have made for this PR, for example, logics for TIMEOUTSTEP and TIMEOUTSTEP_TNIR. Do you think we should close this PR and create other one after we have a final solution?